### DISCUSSION TITLE
Rerun feature runs all tests

### DISCUSSION THREAD

**Discussion by JustinSletcher** [allure-framework/allure2#2067] [Questions & Support] [answered]
Hey! First-time commenter. I want to start off by saying how much our company is enjoying the software. It's been extremely helpful for our whole QA team.

The one feature I've never gotten to work is the rerun feature. Here is how I currently try to use it.

1 - When inspecting the launch results, I select a failed test > bulk actions > rerun
2 - I observe the job executing in Team City with env.ALLURE_JOB_RUN_ID as a new parameter.
3 - It updates the report, but the job runs for 2hours+ which is the full length of the project.

My assumption was that allure would pass in a parameter that I could handle to rerun the test. Eg. The test name or full name, but I don't see anything like this. The documentation that I've read just says to execute the selected tests.

The documentation I've been following is [here](https://docs.qameta.io/allure-testops/project/jobs/run/#rerun-of-tests)

Hopefully, there's something wrong with my setup. 

If it matters, I am able to:
Run tests from TeamCity
Generate reports for the launches
Run tests from Allure Testops

**Comment by baev** [comment]
@JustinSletcher, please specify the framework and Allure integration you're using.

**Reply by JustinSletcher** [reply]
So C#/NUnit/Allure.Nunit/Allure.Commons. I execute the tests with dotnet test.

**Comment by delatrie** [comment, marked as answer]
Hi, @JustinSletcher!

Selective test runs are only available for integrations that support it. Unfortunately, allure-csharp integrations (including Allure.NUnit) doesn't support it yet.

The support will be added in the future. Here is the issue to track the work:
https://github.com/allure-framework/allure-csharp/issues/372

**Reply by JustinSletcher** [reply]
Aha! Thanks. Will keep an eye on it in the future

**Reply by baev** [reply]
Update from 2024: support for selective runs was introduced in 2.10.0
